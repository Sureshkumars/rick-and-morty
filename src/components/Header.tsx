import React from "react";
import { Grid, IconButton } from "@chakra-ui/core";
import { ColorModeSwitcher } from "./ColorModeSwitcher";
import { FaHome } from "react-icons/fa";
import { Link } from "react-router-dom";
export const Header: React.FC = () => {
  return (
    <Grid
      templateColumns="repeat(2, 1fr)"
      gap={6}
      marginLeft="2"
      marginRight="2"
    >
      <Link to="/" title="Return Home">
        <IconButton aria-label="Home Icon" icon={<FaHome />} />
      </Link>
      <ColorModeSwitcher justifySelf="flex-end" />
    </Grid>
  );
};
