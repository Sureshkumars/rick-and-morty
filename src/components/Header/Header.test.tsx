import React from "react";
import { Header } from "./Header";
import { render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";

it("Header matches snapshot", () => {
  const { asFragment }: any = render(
    <MemoryRouter>
      <Route>
        <Header />
      </Route>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
