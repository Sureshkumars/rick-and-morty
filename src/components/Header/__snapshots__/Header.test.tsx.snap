// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-11v0733"
  >
    <a
      href="/"
      title="Return Home"
    >
      <button
        aria-label="Home Icon"
        class="chakra-button css-1kgd4fd"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          />
        </svg>
      </button>
    </a>
    <button
      aria-label="Switch to light mode"
      class="chakra-button css-1mcbq3v"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
